### PR TITLE
Add funtions of Encoder and VIA for redragon/k667

### PR DESCRIPTION
**Pull request by temp4gh**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [X] New feature
- [X] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [X] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/redragon/k667/info.json [hunk 1]
```diff
@@ -6,6 +6,10 @@
         "matrix": [1, 0]
     },
     "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "layer_count": 2
+    },
+
```
[the remark this answers is not in the document]
Comment (zvecr) [reply keyboards/redragon/k667/info.json:12]: It's already been reverted once, so should not be reintroduced.

File: keyboards/redragon/k667/info.json [hunk 2]
```diff
@@ -15,9 +19,12 @@
         "bootmagic": true,
         "console": true,
         "encoder": true,
+
+        "encoder_map": true,
```
Comment (zvecr) [inline keyboards/redragon/k667/info.json:23]: Same reason as via, cannot be enabled at keyboard level 
```suggestion
```

File: keyboards/redragon/k667/keymaps/default/keymap.c [hunk 3]
```diff
@@ -22,6 +22,13 @@ enum layer_names
     _FnLay,
 };
 
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/redragon/k667/keymaps/default/keymap.c:26]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/redragon/k667/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+#include QMK_KEYBOARD_H
+
+enum layer_names
+{
+    _BASE,
+    _FnLay,
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/redragon/k667/keymaps/via/keymap.c:26]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/redragon/k667/keymaps/default/keymap.c
[the remark this answers is not in the document]
Comment (temp4gh) [reply keyboards/redragon/k667/keymaps/default/keymap.c:1]: Many thanks for your suggestion! The file rules.mk have been added to default.

**Review by drashna — APPROVED** [review]

**Pull request merged**